src/components/container/RelativeActivity/RelativeActivity.tsx

```diff
@@ -0,0 +1,111 @@
+import { getDailyDataTypeDefinition } from "../../../helpers/query-daily-data";
+import { ActivityMeter, CardTitle, DateRangeContext, } from "../../presentational";
+import "./RelativeActivity.css"
+import language from "../../../helpers/language";
+import { useInitializeView } from "../../../helpers/Initialization";
+import { RelativeActivityContext } from "../RelativeActivityDayCoordinator/RelativeActivityDayCoordinator";
+import { RelativeActivityDataType, RelativeActivityQueryResult } from "../../../helpers";
+import { queryRelativeActivity } from "../../../helpers/relative-activity";
+
+export interface RelativeActivityProps {
+    dataTypes?: RelativeActivityDataType[];
+    useContext?: boolean;
+    previewState?: "Default";
+    title?: string;
+    innerRef?: React.Ref<HTMLDivElement>
+    date?: Date;
+}
+
+export default function (props: RelativeActivityProps) {
+    let [results, setResults] = useState<{ [key: string]: RelativeActivityQueryResult } | undefined>(undefined);
+    let relativeActivityContext = useContext(RelativeActivityContext);
+
+    let dataTypes: RelativeActivityDataType[] = props.dataTypes || [];
+    if (props.useContext && relativeActivityContext?.dataTypes) {
+        dataTypes = relativeActivityContext.dataTypes;
+    }
+
+    let dateRangeContext = useContext(DateRangeContext);
+    let date = props.date ?? dateRangeContext?.intervalStart ?? startOfDay(new Date());
+
+    function loadData() {
+        function transformResults(results: { [key: string]: { [key: string]: RelativeActivityQueryResult } } | undefined) {
+            if (!results) return;
+            let computedResult: { [key: string]: RelativeActivityQueryResult } = {};
+            dataTypes.forEach(dataType => {
+                computedResult[dataType.dailyDataType] = results[dataType.dailyDataType][getDayKey(date!)];
+            });
+            return computedResult;
+        }
+
+        setResults(undefined);
+        if (props.useContext) {
+            setResults(transformResults(relativeActivityContext!.data));
+            return;
+        }
+        queryRelativeActivity(date!, date!, dataTypes, !!props.previewState).then(results => {
+            setResults(transformResults(results));
+        });
+    }
+
+    useInitializeView(() => {
+        loadData();
+    }, ["externalAccountSyncComplete"], [dateRangeContext, relativeActivityContext, props.date, props.dataTypes], 0);
+
+    if (results == null || !Object.keys(results).length) {
+        return null;
+    }
+
+    function getBarColor(dataType: RelativeActivityDataType, fillPercent: number) {
+        if (dataType.threshold !== undefined && dataType.overThresholdColor && fillPercent > .5) {
+            return dataType.overThresholdColor;
+        }
+        return dataType.color || "var(--mdhui-color-primary)";
+    }
+
+    function getThresholdLabel(dataType: RelativeActivityDataType, threshold: number) {
+        if (!dataTypes.some(d => d.threshold !== undefined)) {
```

Comment:
Rather than the double negative here, I like the following that you used further below:

```dataTypes.every(d => d.threshold === undefined)```

I think those both evaluate to the same thing.

Reply:
Fixed